google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/metrics/BuiltinMetricsTestUtils.java

```diff
@@ -18,33 +18,57 @@
 import static com.google.common.truth.Truth.assertThat;
 
 import com.google.api.core.InternalApi;
+import com.google.common.truth.Correspondence;
 import com.google.protobuf.Timestamp;
 import com.google.protobuf.util.Timestamps;
 import io.opentelemetry.api.common.Attributes;
 import io.opentelemetry.sdk.metrics.data.HistogramPointData;
 import io.opentelemetry.sdk.metrics.data.LongPointData;
 import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.testing.exporter.InMemoryMetricReader;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 import java.util.stream.Collectors;
 import org.junit.Assert;
 
 @InternalApi
 public class BuiltinMetricsTestUtils {
+  private static final Correspondence<MetricData, String> METRIC_DATA_BY_NAME =
+      Correspondence.transforming(MetricData::getName, "MetricData name");
 
   private BuiltinMetricsTestUtils() {}
 
-  public static MetricData getMetricData(Collection<MetricData> allMetricData, String metricName) {
-    List<MetricData> metricDataList =
-        allMetricData.stream()
-            .filter(md -> md.getName().equals(BuiltinMetricsConstants.METER_NAME + metricName))
-            .collect(Collectors.toList());
-    if (metricDataList.size() == 0) {
-      allMetricData.stream().forEach(md -> System.out.println(md.getName()));
+  public static MetricData getMetricData(InMemoryMetricReader reader, String metricName) {
+    String fullMetricName = BuiltinMetricsConstants.METER_NAME + metricName;
+    Collection<MetricData> allMetricData = Collections.emptyList();
+    Optional<MetricData> filterdMetricData = Optional.empty();
+
+    // Fetch the MetricData with retries
+    for (int attemptsLeft = 10; attemptsLeft >= 0; attemptsLeft--) {
```

Comment:
nit: not sure if the semantics of "attemptsLeft = 10" imply it should run for 10 times or 11, but the loop runs 11 times in case it's the former case.

Reply:
fixed